### PR TITLE
fix(getters/shrex): removing implicit context assumption

### PR DESCRIPTION
**Pull request by distractedm1nd**
## Overview
Prevents infinite processing during availability calls

### CODE REVIEW

File: share/getters/shrex.go [hunk 1]
```diff
@@ -34,6 +34,11 @@ func (sg *ShrexGetter) GetShare(ctx context.Context, root *share.Root, row, col
 
 func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.ExtendedDataSquare, error) {
 	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+		}
 		peer, setStatus, err := sg.peerManager.Peer(ctx, root.Hash())
```
Comment (Wondertan) [inline share/getters/shrex.go:42]: AFAICS, Peer method handles context correctly, and usually, such ctx check means the call down the stack is blocking. Why do we need to do it?

File: share/getters/shrex.go [hunk 2]
```diff
@@ -34,6 +34,11 @@ func (sg *ShrexGetter) GetShare(ctx context.Context, root *share.Root, row, col
 
 func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.ExtendedDataSquare, error) {
 	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+		}
```
Comment (Wondertan) [inline share/getters/shrex.go:41]: alternative and shorter form
```suggestion
	for ctx.Err == nil {
```

File: share/getters/shrex.go (hunk 1 above)
Comment (distractedm1nd) [reply share/getters/shrex.go:42]: We need it in the case that there is a peer in the pool. `Peer` returns a peer before it checks the context in this case.

File: share/getters/shrex.go (hunk 2 above)
Comment (distractedm1nd) [reply share/getters/shrex.go:41]: based

**Review by walldiss — APPROVED** [review]

File: share/getters/shrex.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ func (sg *ShrexGetter) GetShare(ctx context.Context, root *share.Root, row, col
 }
 
 func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.ExtendedDataSquare, error) {
-	for {
+	for ctx.Err() != nil {
```
Comment (renaynay) [inline share/getters/shrex.go:36]: huh? shouldn't it be the other way around?  and even so, can we stick w wrapping w select pattern pls?

**Review by renaynay — CHANGES_REQUESTED** [review]

File: share/getters/shrex.go (hunk 3 above)
Comment (distractedm1nd) [reply share/getters/shrex.go:36]: yes

File: share/getters/shrex.go (hunk 2 above)
Comment (renaynay) [reply share/getters/shrex.go:41]: we use select pattern everywhere else so lets stick to it but im fine w shorter form if we unify later on

**Review by renaynay — APPROVED** [review]

**Pull request merged**